include/boost/gil/extension/numeric/convolve.hpp

```diff
@@ -355,6 +356,77 @@ void convolve_cols_fixed(
         transposed_view(src_view), kernel, transposed_view(dst_view), option);
 }
 
+namespace detail
+{
+
+template <typename SrcView, typename DstView, typename Kernel>
+void convolve_2d_impl(SrcView const& src_view, DstView const& dst_view, Kernel const& kernel)
+{
+    int flip_ker_row, flip_ker_col, row_boundary, col_boundary;
+    float aux_total;
+    for (std::ptrdiff_t view_row = 0; view_row < src_view.height(); ++view_row)
+    {
+        for (std::ptrdiff_t view_col = 0; view_col < src_view.width(); ++view_col)
+        {
+            aux_total = 0.0f;
+            for (std::size_t kernel_row = 0; kernel_row < kernel.size(); ++kernel_row)
+            {
+                flip_ker_row = kernel.size() - 1 - kernel_row;      // row index of flipped kernel
+
+                for (std::size_t kernel_col = 0; kernel_col < kernel.size(); ++kernel_col)
+                {
+                    flip_ker_col = kernel.size() - 1 - kernel_col; // column index of flipped kernel
+
+                    // index of input signal, used for checking boundary
+                    row_boundary = view_row + (kernel.center_vertical() - flip_ker_row);
+                    col_boundary = view_col + (kernel.center_horizontal() - flip_ker_col);
+
+                    // ignore input samples which are out of bound
+                    if (row_boundary >= 0 && row_boundary < src_view.height() &&
+                        col_boundary >= 0 && col_boundary < src_view.width())
+                    {
+                        aux_total +=
+                            src_view(col_boundary, row_boundary) *
+                            kernel.at(flip_ker_row, flip_ker_col);
+                    }
+                }
+            }
+            dst_view(view_col, view_row) = iround(aux_total);
+        }
+    }
+}
+
+} //namespace detail
+
+/*!
```

Comment:
In future, please prefer the Doxygen's C++ style `///` instead of Javadoc or Qt styles.

Shortly, a recommended structure of a Doxygen-style function comment is this:

```
/// \brief Short summary, one sentence
/// ...detailed documentation ...
/// \tparam T1 - template parameter...
/// \tparam T2 - template parameter...
/// \param p1 - function parameter...
/// \param p2 - function parameter...
```